File: cpp-std/demo01a01-hello.cpp
/*
HELLO WORLD VERSION MULTITHREADING
Version A01: Using functions
*/


#include <iostream>
#include <thread>
using namespace std;



void doTask() {
    cout << "Hello from example thread" << endl;
}



int main() {
    std::thread th(doTask);

    cout << "Hello from main thread" << endl;

    th.join();
    return 0;
}

File: csharp/demo/demo01a01-hello.cs
/*
 * HELLO WORLD VERSION MULTITHREADING
 * Version A01: Using functions
 */
using System;
using System.Threading;



class Demo01A01 : IRunnable
{
    public void run()
    {
        Thread th = new Thread(lorem);
        th.Start();
        Console.WriteLine("Hello from main thread");
    }

    private void lorem()
    {
        Console.WriteLine("Hello from Lorem thread");
    }
}
